src/clj/clojurians_log/components/parser.clj

```diff
@@ -0,0 +1,23 @@
+(ns clojurians-log.components.parser
+  (:require [clojure.string :as str]))
+
+(defn- extract-texts
+  [messages]
+  (let [texts (map :message/text messages)]
+    (clojure.string/join " " texts)))
+
+(defn- parse-users
+  [texts]
+  (re-seq #"(?<=<@).*?(?=>)" texts))
+
+(defn extract-user-ids
+  "returns all the user/slack-ids from a string"
+  [messages]
+  (parse-users (extract-texts messages)))
```

Comment:
You're joining all these messages into a string so you can let the regex do its thing, I would avoid building up that big string.

```
(mapcat (comp parse-users :message/text) messages)
```

You also don't care about duplicates, which means maybe this should be a set, which means you'll be using `into`, which means you might as well use a transducer.

```
(into #{} (comp
           (map :message/text)
           (map parse-users)
           cat)
      messages)
```